### PR TITLE
Merge git-init image source code into git-clone repo

### PR DESCRIPTION
**Pull request by ywluogg**
Merge git-init image source code into git-clone repo.

Previously git-init image source code is in Tekton Pipeline repository and is removed by this https://github.com/tektoncd/pipeline/pull/6417

It's agreed by the community that the source code will be merged into git-clone repo.

#2

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

**Review by afrittoli — COMMENTED** [review]
Thanks @ywluogg!
I'm definitely in favour of this code being merged.

We have a `task` folder.  if the intention is to have the same folder structure for all repos in this org, perhaps it should be called `tasks` instead, as it may be more than one.

I think it would be really useful to have a root level README file, that explains what's the content of each folder and how to use it (either directly or link to folder README).

**Comment by vdemeester** [comment]
> Thanks @ywluogg! I'm definitely in favour of this code being merged.
> 
> We have a `task` folder. if the intention is to have the same folder structure for all repos in this org, perhaps it should be called `tasks` instead, as it may be more than one.
> 
> I think it would be really useful to have a root level README file, that explains what's the content of each folder and how to use it (either directly or link to folder README).

Those 2 could be done in a follow-up I think 👼🏼

File: task/git-clone/git-clone.yaml [hunk 1]
```diff
@@ -1,238 +1,238 @@
 apiVersion: tekton.dev/v1beta1
```
[the remark this answers is not in the document]
Comment (ywluogg) [reply task/git-clone/git-clone.yaml:1]: This is coming from my yamllint. I can change back to the original one if you are concerned that this PR is not intended to lint the yaml

[the remark this answers is not in the document]
Comment (afrittoli) [reply task/git-clone/git-clone.yaml:1]: That's fine, it only makes review harder, if you can confirm that there is no actual change than it's fine

**Comment by ywluogg** [comment]
> > > > Thanks @ywluogg! I'm definitely in favour of this code being merged.
> > > > We have a `task` folder. if the intention is to have the same folder structure for all repos in this org, perhaps it should be called `tasks` instead, as it may be more than one.
> > > > I think it would be really useful to have a root level README file, that explains what's the content of each folder and how to use it (either directly or link to folder README).
> > > 
> > > 
> > > Those 2 could be done in a follow-up I think 👼🏼
> > 
> > 
> > +1 but I wanted to share that https://github.com/tektoncd/community/blob/main/teps/0110-decouple-catalog-organization-and-reference.md and https://github.com/tektoncd/community/blob/main/teps/0003-tekton-catalog-organization.md both set the policy to have the naming to be {resouce_type}, not {resource_type}s, for the decentralized and centralized catalogs. It does seem like a policy we need some consensus on.
> > I'm okay to add the root level README later.
> 
> Thanks for the clarification. Should the folder be `image` instead of `images` then? Just a NIT, it's fine to merge as it is.

Good point! I just refreshed this to align the naming

File: task/git-clone/git-clone.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ywluogg) [reply task/git-clone/git-clone.yaml]: Ah good point. I will move the linting as a followup PR

**Comment by vdemeester** [comment]
I've created an issue to follow-up on this https://github.com/tektoncd-catalog/git-clone/issues/5.
I'll go ahead and merge this to unblock things.

**Pull request merged**